.github/workflows/dependency-check.yml

```diff
@@ -0,0 +1,35 @@
+name: Check dependencies using safety
+
+on:
+  push:
+    paths:
+      - requirements.txt
+      - .github/workflows/dependency-check.yml
+    branches: [ main ]
+  pull_request:
+    paths:
+      - requirements.txt
+      - .github/workflows/dependency-check.yml
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+
+jobs:
+  dependency-check:
+    runs-on: ubuntu-latest
+    name: Safety
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Set up Python environment
+      uses: actions/setup-python@v3
+      with:
+        python-version: 3.9
```

Comment:
I'm at 3.9.12, is this something we need to align here? I cannot run 3.9 locally due to some Apple Silicon stuff

Reply:
I think 3.9 or 3.9.12 should not really matter?

Reply:
It does for Apple Silicon. Is there a cost to you for using 3.9.12? There is a cost to me using 3.9 since the local docker image won't build for me then.

Reply:
Nope no problem! Will change it